libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -1303,10 +1319,20 @@ MAV_MISSION_RESULT AP_Mission::mavlink_int_to_mission_cmd(const mavlink_mission_
         break;
 
     case MAV_CMD_DO_GIMBAL_MANAGER_PITCHYAW:
-        cmd.content.gimbal_manager_pitchyaw.pitch_angle_deg = packet.param1;
-        cmd.content.gimbal_manager_pitchyaw.yaw_angle_deg = packet.param2;
-        cmd.content.gimbal_manager_pitchyaw.pitch_rate_degs = packet.param3;
-        cmd.content.gimbal_manager_pitchyaw.yaw_rate_degs = packet.param4;
+        if (!isnan(packet.param1)) {
+            cmd.content.gimbal_manager_pitchyaw.pitch_angle_deg = packet.param1;
+        }
+        else                                                            // Sets the angles out of range, 
+            cmd.content.gimbal_manager_pitchyaw.pitch_angle_deg = 100;  // causes start_command_do_gimbal_manager_pitchyaw to skip setting angles
+        if (!isnan(packet.param2)) {
+            cmd.content.gimbal_manager_pitchyaw.yaw_angle_deg = packet.param2;
+        }        
+        if (!isnan(packet.param3)) {
+            cmd.content.gimbal_manager_pitchyaw.pitch_rate_degs = packet.param3;
+        }
+        if (!isnan(packet.param4)) {
+            cmd.content.gimbal_manager_pitchyaw.yaw_rate_degs = packet.param4;
+        }
```

Comment:
I don't think you can do this here, it needs to be done in the Mount driver.

Setting these values to `NaN` means keep the current value. However, if these are `NaN`, the `cmd` values will be set to `0`.

Reply:
A possible solution: since the gimbal_manager_pitchyaw_Command struct is size 7 bytes, and mission structs can contain 10 bytes of data, I can use a free byte to set which params are NaN where one bit corresponds to one param. I can pass it in to start_command_do_gimbal_manager_pitchyaw. That way we can know which param is NaN.

Reply:
I think you can remove this block completely. The `cmd` values are already checked for `NaN` in [`AP_Mission::start_command_do_gimbal_manager_pitchyaw()`](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Mission/AP_Mission_Commands.cpp#L289) and [`AP_Mount::handle_command_do_gimbal_manager_pitchyaw()`](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Mount/AP_Mount.cpp#L346).

Reply:
Hey @rmackay9, should [`AP_Mission::start_command_do_gimbal_manager_pitchyaw()`](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Mission/AP_Mission_Commands.cpp#L263) just call [`AP_Mount::handle_command_do_gimbal_manager_pitchyaw()`](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Mount/AP_Mount.cpp#L313), rather than duplicating the functionality?

Reply:
> I think you can remove this block completely. The `cmd` values are already checked for `NaN` in [`AP_Mission::start_command_do_gimbal_manager_pitchyaw()`](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Mission/AP_Mission_Commands.cpp#L289) and [`AP_Mount::handle_command_do_gimbal_manager_pitchyaw()`](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Mount/AP_Mount.cpp#L346).

I am not an expert in ArduPilot source code but my understanding is that Mission commands are placed in EEPROM and then run as per their sequence. uint8_t and int8_t data types are used to encode the data from the commands rather than floats as we are limited to 10 bytes for storing command parameters. The issue I am having is that NaN is a value specific to floats, setting an int to NaN throws an arithmetic exception.

Reply:
Yeah, you're right. Sorry, I forgot that the Mission Commands were reduced in size using ints. 

Re. your earlier solution:
> A possible solution: since the gimbal_manager_pitchyaw_Command struct is size 7 bytes, and mission structs can contain 10 bytes of data, I can use a free byte to set which params are NaN where one bit corresponds to one param.

That would work, but rather than waste another byte, I'd do something like the following:
```
cmd.content.gimbal_manager_pitchyaw.pitch_angle_deg = isnan(packet.param1) ? INT8_MAX : packet.param1;
```
and then do the inverse in `AP_Mission::start_command_do_gimbal_manager_pitchyaw()`.

> `AP_Mission::start_command_do_gimbal_manager_pitchyaw()` checks for NaN but the cmd.content.gimbal_manager_pitchyaw struct cannot be NaN in the first place.

Yeah, that is broken.

Reply:
Hey Nick,
I will get in touch with you on Discord. Perhaps we can make a thread about this in one of the development channels in the ArduPilot Discord? My username is: trolledbypro

Reply:
Implemented the changes in my new commit